openedx/core/djangoapps/discussions/models.py

```diff
@@ -0,0 +1,55 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from jsonfield import JSONField
+from opaque_keys.edx.django.models import LearningContextKeyField
+
+from openedx.core.djangoapps.config_model_utils.models import CourseAppConfigOptionsModel
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+class DiscussionConfig(CourseAppConfigOptionsModel):
+    """
+    Configuration model to store configuration for Discussions applications.
+    """
+
+    provider = models.CharField(
+        blank=False,
+        db_index=True,
+        max_length=100,
+        verbose_name=_("Discussion provider"),
+        help_text=_("The discussion tool/provider."),
+    )
+    config = JSONField(
+        blank=False,
+        default={},
+        help_text=_("The discussion configuration data that can be user visible."),
+    )
+
+
+class LearningContextDiscussionConfig(models.Model):
+    """
+    Associates each learning context with a :class:`DiscussionConfiguration` via the config_key.
```

Comment:
Nit: I recommend writing "each learning context **(e.g. each course run)** with a..." to make it clear that generally "learning context" = "course"